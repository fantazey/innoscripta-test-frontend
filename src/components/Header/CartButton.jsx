import React from "react";
import {NavLink} from "react-router-dom";

export default class CartButton extends React.Component {
  render() {
    return <div className={"mr-1 bg-warning"} style={{'borderRadius': '0.4em'}}>
      <NavLink exact
               className='nav-link'
               activeClassName='active'
               to={"/cart"}>
        <span>Cart</span>
        <span className={"pl-2"}>{this.props.cartPrice}</span>
      </NavLink>
    </div>;
  }
}