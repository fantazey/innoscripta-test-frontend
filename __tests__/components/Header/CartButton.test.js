import CartButton from '../../../src/components/Header/CartButton';

describe('CartButton', () => {
  let wrapper;

  beforeEach(() => {
    const price = 12;
    wrapper = shallow(
      <CartButton cartPrice={price} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Button label should be translated', () => {
    const label = wrapper.find('span').first();
    expect(label.text().startsWith('_')).toEqual(true);
  });
});
