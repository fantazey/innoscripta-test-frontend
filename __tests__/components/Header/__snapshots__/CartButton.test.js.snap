// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartButton should match snapshot 1`] = `
<div
  className="d-flex"
>
  <div
    className="mr-1 bg-warning"
    style={
      Object {
        "borderRadius": "0.4em",
      }
    }
  >
    <NavLink
      activeClassName="active"
      className="nav-link"
      exact={true}
      to="/cart"
    >
      <span>
        _cart
      </span>
      <span
        className="pl-2"
      >
        <CurrencyPrice
          price={12}
        />
      </span>
    </NavLink>
  </div>
</div>
`;
